packages/asset/contracts/Asset.sol

```diff
@@ -168,6 +189,15 @@ contract Asset is
                 revealHashes[i]
             );
             amounts[i] = 1;
+            if (ipfsStringUsed[assetUris[i]] != 0) {
+                require(
+                    ipfsStringUsed[assetUris[i]] == tokenIds[i],
+                    "ipfs already used"
+                );
+            } else {
+                ipfsStringUsed[assetUris[i]] = tokenIds[i];
+            }
```

Comment:
When we reveal an asset we may reveal exactly same copy, so the ipfs hash will be the same as other already existing tokens where we just increase the copies amount.

Reply:
Same for the bridge if a person bridge 1 copy of an asset from L1 and then try to bridge another copy it might be the same asset so the ipfs hash will be the same.